Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/BoundsControl/BoundsControl.cs

```diff
@@ -0,0 +1,1155 @@
+
+        private readonly List<IMixedRealityInputSource> touchingSources = new List<IMixedRealityInputSource>();
+      
+        private List<IMixedRealityController> sourcesDetected;
+      
+        // Current axis of rotation about the center of the rig root
+        private Vector3 currentRotationAxis;
+
+        // Scale of the target at the beginning of the current manipulation
+        private Vector3 initialScaleOnGrabStart;
+
+        // Position of the target at the beginning of the current manipulation
+        private Vector3 initialPositionOnGrabStart;
+
+        // Point that was initially grabbed in OnPointerDown()
+        private Vector3 initialGrabPoint;
+
+        // Current position of the grab point
+        private Vector3 currentGrabPoint;
+
+        private TransformScaleHandler scaleHandler;
+
+        // Grab point position in pointer space. Used to calculate the current grab point from the current pointer pose.
+        private Vector3 grabPointInPointer;
+
+        // Corner opposite to the grabbed one. Scaling will be relative to it.
+        private Vector3 oppositeCorner;
+
+        // Direction of the diagonal from the opposite corner to the grabbed one.
+        private Vector3 diagonalDir;
+
+        private HandleType currentHandleType;
+
+        // The size, position of boundsOverride object in the previous frame
+        // Used to determine if boundsOverride size has changed.
+        private Bounds prevBoundsOverride = new Bounds();
+
+        // True if this game object is a child of the Target one
+        private bool isChildOfTarget = false;
+        private static readonly string rigRootName = "rigRoot";
+
+        // Cache for the corner points of either renderers or colliders during the bounds calculation phase
+        private static List<Vector3> totalBoundsCorners = new List<Vector3>();
+
+        //public class BoundsChangedEvent: UnityEvent<Vector3[]> {}
+        //internal BoundsChangedEvent boundsUpdated = new BoundsChangedEvent();
+        private UnityEvent boundsUpdated = new UnityEvent();
+
+        private Vector3[] boundsCorners = new Vector3[8];
+        public Vector3[] BoundsCorners { get; private set; }
+
+        #endregion
+
+        #region public Properties
+
+        /// <summary>
+        /// The collider reference tracking the bounds utilized by this component during runtime
+        /// </summary>
+        public BoxCollider TargetBounds { get; private set; }
+
+        // TODO Review this, it feels like we should be using Behaviour.enabled instead.
+        private bool active = false;
+        public bool Active
+        {
+            get
+            {
+                return active;
+            }
+            set
+            {
+                if (active != value)
+                {
+                    active = value;
+                    rigRoot?.gameObject.SetActive(value);
+                    ResetVisuals();
+
+                    if (active)
```

Comment:
nit: curly braces. Also there's lots of extra whitespace here and elsewhere. Probably worth doing a pass to clean that up.